test/m3u8.test.js

```diff
@@ -1,646 +1,650 @@
-(function(window, undefined) {
-  var
-    //manifestController = this.manifestController,
-    m3u8 = window.videojs.m3u8,
-    ParseStream = m3u8.ParseStream,
-    parseStream,
-    LineStream = m3u8.LineStream,
-    lineStream,
-    Parser = m3u8.Parser,
-    parser;
-
-  /*
-    M3U8 Test Suite
-  */
-
-  QUnit.module('LineStream', {
-    setup: function() {
-      lineStream = new LineStream();
+import {ParseStream, LineStream, Parser} from '../src/m3u8';
+import QUnit from 'qunit';
+import testDataExpected from './data/expected.js';
+import testDataManifests from './data/manifests.js';
+
+QUnit.module('LineStream', {
+  beforeEach() {
+    this.lineStream = new LineStream();
```

Comment:
i was running into issues in a few of these files related to tests not cleaning up after themselves when they modify the global, or redeclaring the global in their scope (and the linter complaining that the variable was already declared). I have also had problems with tests in the past that use globals which caused future tests to fail/pass making it very confusing to debug. the generator also generated several tests that all used this. so I assumed that was the way we probably wanted to go.

That being said I am not opposed to reverting it to not us `this.`. It just seemed easier to reason about tests if there was no chance that scope would leak.